dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -194,6 +209,62 @@ def profile(self):
             )
         return profile
 
+    def diff(self, other, options=None):
+        """
+        Finds the difference between 2 compilers and returns the report
+        
+        :param other: profile compiler finding the difference with this one.
+        :type other: ColumnPrimitiveTypeProfileCompiler
+        :return: difference of the profiles
+        :rtype: dict
+        """
+        # Call super for compiler instance check
+        super().diff(other)
+
+        # Initialize profile diff dict
+        profile = {
+            "data_type_representation": dict(),
+        }
+
+        # Iterate through self for primitive profiles
+        matched_profile1 = None
+        for key, profiler in self._profiles.items():
+            # Find data_type
+            if matched_profile1 is None and profiler.data_type_ratio == 1.0:
+                matched_profile1 = key
```

Comment:
Do you think we should have something in the base which gets this auto? I imagine `def profile` would use it too

Reply:
e.g. `self.profile_type` or something?

Reply:
Separated out into its own function, but this is the only compiler that will use it